forge-gui/res/cardsfolder/upcoming/a_premonition_of_your_demise.txt

```diff
@@ -0,0 +1,10 @@
+Name:A Premonition of Your Demise
+ManaCost:no cost
+Types:Scheme
+T:Mode$ SetInMotion | ValidCard$ Card.Self | Execute$ TrigRepeatEach | TriggerZones$ Command | TriggerDescription$ When you set this scheme in motion, for each card type among permanents your opponents control, you may put a permanent card of that type from your hand onto the battlefield.
+SVar:DBDig:DB$ Dig | DigNum$ 2 | Reveal$ True | ChangeNum$ All | ChangeValid$ Card | DestinationZone$ Hand | RememberChanged$ True | SubAbility$ DBImmediateTrigger
+SVar:DBImmediateTrigger:DB$ ImmediateTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card.nonLand | Execute$ TrigDamage | SpellDescription$ When you reveal one or more nonland cards this way, this scheme deals damage equal to their total mana value to any target.
+SVar:TrigDamage:DB$ DealDamage | ValidTgts$ Any | NumDmg$ X | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Remembered.nonLand$SumCMC

```

Comment:
was this line tested?

Reply:
No, I mean, technically nonLand is not even needed there since lands have mana value 0, it was just futureproofing.

Reply:
well still seems like invalid syntax...?

Reply:
ok, so how should it look?

Reply:
Hmn I don't think you can filter a value with Defined via Valid and then apply another function
It would have to be Count$Valid I guess?

Reply:
I think I'm a bit lost... is this right?

Reply:
nope, anyway now most cases should work 🤷‍♂️

Reply:
Thanks :)